plugin/trino-hive/src/test/java/io/trino/plugin/hive/TestHive2OnDataLakeOverProxy.java

```diff
@@ -0,0 +1,67 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.trino.plugin.hive;
+
+import com.google.common.collect.ImmutableMap;
+import io.trino.plugin.hive.containers.HiveHadoop;
+import io.trino.testing.DistributedQueryRunner;
+import io.trino.testing.containers.Httpd;
+import io.trino.testing.containers.Minio;
+import org.testcontainers.containers.Network;
+
+import java.util.Map;
+
+import static io.trino.testing.containers.Httpd.builderForHttpProxy;
+
+public class TestHive2OnDataLakeOverProxy
+        extends BaseTestHiveOnDataLake
```

Comment:
How many, and which tests do we run with S3 HTTP proxy?

Reply:
I think this is the first one in trino repository. And in this particular case all underlining tests in class will be using S3 (MinIO) over proxy.

Reply:
I understand, but the tests actually invoked look like a random selection. For example flush procedure tests.

Reply:
Yes that's true. Cause behind my decision is that. I needed:
- Hive Data Lake set up
- test's utilising complex operations on S3 (MinIO). 
BaseTestHiveOnDataLake test seems to be perfect fit aside cache flush (which are pretty fast so shouldn't be a big deal).
We could ofcourse sub-select some tests and either replicate them or make separate base class. 
But I'm not sure if it's worth. In current setup it's a natural place to add new use cases for hive data lake. And we will have a proof that all of those use cases would work over proxy as well (which i agree might be an overkill 🤷‍♂️ )